bip-p2qrh.mediawiki

```diff
@@ -50,26 +50,40 @@ The following table is non-exhaustive, but meant to be informative to the averag
+Not having public keys exposed on-chain is a major constraint for quantum security, otherwise funds would need to be spent to new addresses on a regular basis in order to prevent the possibility of a "long-range CRQC attack" recovering the key behind high value addresses. A long-range quantum attack can be considered one performed with chain data, such as that from a used address or one encoded in a spend script. A "short-range quantum attack" would be one done on keys in the mempool, which is seen as impractical given transaction throughput and the block time. The more value being sent, the higher the fee should be set, in order to commit the transaction to the chain as soon as possible, thus making useless the public key revealed by spending a UTXO.
+
+Support for FALCON will also be introduced. While its signatures are roughly 4x larger than SQIsign signatures and 20x larger than Schnorr signatures, it will be necessary to have a fallback algorithm should a flaw be found in SQIsign, which is much newer and experimental. FALCON is a more conservative approach to applied lattice cryptography than SQIsign, and its use is approved by NIST. However, even SQIsign signatures are roughly 5x larger than Schnorr signatures. This means, to maintain present transaction throughput, an increase in the witness discount will likely be desired in a QuBit soft fork. That will be specified in a future QuBit BIP.
+
+An increase in the witness discount must not be taken lightly, and must be resistant to applications that might take advantage of this discount such as storage of arbitrary data (commonly known as "inscriptions"), inflating chain state on nodes without a corresponding increase in economic activity. This would not only impact node runners, but those with inscriptions would also have the scarcity of their non-monetary assets affected.
+
+To address this, very specific rules will be applied to spending from the witness stack in SegWit v3 outputs. A fixed signature size will be necessary for spending the output, and the output must be spendable to be considered valid within node consensus. A fixed signature size will also be helpful in disambiguating between signature types without an additional version byte, as SQIsign signatures are substantially smaller than FALCON signatures, and vice versa, and so the correct signature algorithm can be inferred. The public key and signature will be pushed separately to the witness stack. Multiple signatures can be included in order to support multisig applications.
```

Comment:
It is not completely clear what "this" is referring to in the first sentence.  Also, I would break the third sentence into something more palatable like this:

> A fixed signature size will also be helpful in disambiguating between signature types without an additional version byte, as SQIsign signatures are substantially smaller than FALCON signatures, and vice versa.  Consequently, the correct signature algorithm can be inferred.

Reply:
```suggestion
To address this, very specific rules will be applied to spending from the witness stack in SegWit v3 outputs. A fixed signature size will be necessary for spending the output, and the output must be spendable to be considered valid within node consensus. A fixed signature size will also be helpful to disambiguate between signature types without an additional version byte, as SQIsign signatures are substantially smaller than FALCON signatures, and vice versa. Consequently, the correct signature algorithm can be inferred. The public key and signature will be pushed separately to the witness stack. Multiple signatures can be included in order to support multisig applications.
```

How's that?

Reply:
Looks better.  What does the "this" refer to in the 1st sentence?  Can we make that explicit?

Reply:
What does the "vice versa" part above imply?

Reply:
I've updated it to make it more clear on both those points. Good call.